### PR TITLE
Updating L1 Track Matched Objects

### PR DESCRIPTION
**Pull request by suchandradutta**
a few access methods added in LTkEmParticle; 
L1TkPrimaryVertex now defined within namespace l1t;
 bug fix in L1TkElectronTrackMatchAlgo ; 
updated in L1TkObjectAnalyzer using template  methods and switched of TkMuon analysis for the moment

### CODE REVIEW

File: L1Trigger/L1TTrackMatch/src/L1TkElectronTrackMatchAlgo.cc [hunk 1]
```diff
@@ -91,7 +91,7 @@ namespace L1TkElectronTrackMatchAlgo {
 	double rperp = 129.0;
 	double zface =  sqrt( cos( theta ) * cos( theta ) /
 			     ( 1 - cos( theta ) * cos( theta ) ) * 
-			     rperp * rperp ) * fabs( eta ) / eta;  
+			      rperp * rperp );
```
Comment (rekovic) [inline L1Trigger/L1TTrackMatch/src/L1TkElectronTrackMatchAlgo.cc:94]: @suchandradutta 
Removing `* fabs( eta ) / eta` leaves the `zface` always positive. 
This does not effectively change anything as the subsequent and only use of `zface` is in line below `zface * zface`.

Comment (rekovic) [reply L1Trigger/L1TTrackMatch/src/L1TkElectronTrackMatchAlgo.cc:94]: Is this the intended fix in L1TkElectronMatchAlgo that you mentioned in the introduction?

Comment (suchandradutta) [reply L1Trigger/L1TTrackMatch/src/L1TkElectronTrackMatchAlgo.cc:94]: Yes this is the fix I referred. Algorithm wise there is no change as zface *  zface is used but the function was not protected for eta=0

Comment (rekovic) [reply L1Trigger/L1TTrackMatch/src/L1TkElectronTrackMatchAlgo.cc:94]: Ok. Thanks.

**Pull request merged**